src/app/components/itenary/itenary.component.ts

```diff
@@ -14,26 +13,50 @@ import {
   matAdd,
 } from '@ng-icons/material-icons/baseline';
 import { NgIconComponent, provideIcons } from '@ng-icons/core';
+import { AddTripFormComponent } from '../shared/add-trip-form/add-trip-form.component';
+import { ActivatedRoute } from '@angular/router';
+import { DayComponent } from '../day/day.component';
+import { DayInterface, TripInterface } from '../../models';
+import createDays from '../../utils/createDays-utils';
+import { TripState } from '../../store/state';
+import { selectSelectedTrip } from '../../store/selectors';
 
 @Component({
   selector: 'app-itenary',
   standalone: true,
+  templateUrl: './itenary.component.html',
+  styleUrl: './itenary.component.css',
+  viewProviders: [provideIcons({ matEdit, matLocationOn, matAdd })],
   imports: [
     AsyncPipe,
     EventComponent,
     NzTabsModule,
     NzRadioModule,
     NzInputNumberModule,
     NgIconComponent,
+    AddTripFormComponent,
+    DayComponent,
   ],
-  templateUrl: './itenary.component.html',
-  styleUrl: './itenary.component.css',
-  viewProviders: [provideIcons({ matEdit, matLocationOn, matAdd })],
 })
-export class ItenaryComponent {
-  selectedTrips$ = this.store.select(selectTrips);
+export class ItenaryComponent implements OnInit {
+  selectedTrip$ = this.store.select(selectSelectedTrip);
+
+  tripId = '';
+  daysBetween: DayInterface[] = [];
+  trip: TripInterface | null = null;
 
-  constructor(private store: Store<AppState>) {
+  constructor(private store: Store<TripState>, private route: ActivatedRoute) {
     this.store.dispatch(getTrips());
   }
+
+  ngOnInit(): void {
+    this.tripId = this.route.snapshot.params['tripId'];
+
+    this.selectedTrip$.subscribe((trip) => {
```

Comment:
There's two options to unsubscribe to this observable: 

1. `takeUntilDestroyed` from `rxjs-interop`: `this.selectedTrip$.pipe(takeUntilDestroyed()).subscribe(...)`
2. unsubscribe in `ngOnDestroy`: this.selectedTrip$.unsubscribe()

I recommend the first one.